### PR TITLE
Add btn to add and refresh llm

### PR DESCRIPTION
**Pull request by Bhavyajain21**
/claim #445

### CODE REVIEW

File: src/components/Chat/StartChat.tsx [hunk 1]
```diff
@@ -41,6 +50,14 @@ const StartChat: React.FC<StartChatProps> = ({ defaultModelName, handleNewChatMe
     'Based on what I wrote last week, which tasks should I focus on this week?',
   ])
 
+  const [isSettingsModalOpen, setSettingsModalOpen] = useState(false)
+  const [activeTab, setActiveTab] = useState<SettingsTab>(SettingsTab.GeneralSettingsTab)
```
[the remark this answers is not in the document]
Comment (Bhavyajain21) [reply src/components/Chat/StartChat.tsx]: Fixed, implemented a custom hook

File: src/components/Chat/StartChat.tsx [hunk 2]
```diff
@@ -145,18 +171,25 @@ const StartChat: React.FC<StartChatProps> = ({ defaultModelName, handleNewChatMe
...
-                    </SelectTrigger>
-                    <SelectContent>
-                      {llmConfigs.map((llm) => (
-                        <SelectItem key={llm.modelName} value={llm.modelName}>
-                          {llm.modelName}
-                        </SelectItem>
-                      ))}
-                    </SelectContent>
-                  </Select>
+                  {llmConfigs.length === 0 ? (
```
[the remark this answers is not in the document]
Comment (Bhavyajain21) [reply src/components/Chat/StartChat.tsx]: fixed!

**Comment by Bhavyajain21** [comment]
@joseplayero please review!

File: src/components/Chat/StartChat.tsx [hunk 3]
```diff
@@ -145,18 +169,13 @@ const StartChat: React.FC<StartChatProps> = ({ defaultModelName, handleNewChatMe
...
-                      ))}
-                    </SelectContent>
-                  </Select>
+                  <LLMSelectOrButton
+                    llmConfigs={llmConfigs}
+                    selectedLLM={selectedLLM}
+                    handleLLMChange={handleLLMChange}
+                    openLLMSettings={openLLMSettings}
+                  />
+                  <FiRefreshCw onClick={refreshLLMConfigs} className="ml-2 cursor-pointer" />
```
[the remark this answers is not in the document]
Comment (joseplayero) [reply src/components/Chat/StartChat.tsx:178]: also it should be smaller and less white, more gray

File: src/components/Chat/StartChat.tsx [hunk 4]
```diff
@@ -106,6 +121,15 @@ const StartChat: React.FC<StartChatProps> = ({ defaultModelName, handleNewChatMe
     })
   }
 
+  const refreshLLMConfigs = async () => {
```
Comment (joseplayero) [inline src/components/Chat/StartChat.tsx:124]: this func should also go in llmselectorbutton.tsx

**Review by joseplayero — CHANGES_REQUESTED** [review]

File: src/components/Chat/StartChat.tsx [hunk 5]
```diff
@@ -24,12 +23,23 @@ import exampleAgents from './ChatConfigComponents/exampleAgents'
...
+import SettingsModal from '../Settings/Settings'
+import useModalState from '../Settings/LLMSettings/hooks/use-modal-state'
+import LLMSelectOrButton from '../Settings/LLMSettings/LLMSelectOrButton'
 
 interface StartChatProps {
   defaultModelName: string
   handleNewChatMessage: (userTextFieldInput?: string, chatFilters?: AgentConfig) => void
 }
 
+enum SettingsTab {
```
[the remark this answers is not in the document]
Comment (Bhavyajain21) [reply src/components/Chat/StartChat.tsx]: Done!

File: src/components/Settings/LLMSettings/hooks/use-modal-state.ts [hunk 6]
```diff
@@ -0,0 +1,32 @@
+import { useState } from 'react'
```
[the remark this answers is not in the document]
Comment (Bhavyajain21) [reply src/components/Settings/LLMSettings/hooks/use-modal-state.ts]: Done!

File: src/components/Chat/StartChat.tsx [hunk 7]
```diff
@@ -273,6 +292,7 @@ const StartChat: React.FC<StartChatProps> = ({ defaultModelName, handleNewChatMe
           </div>
         </div>
       </div>
+      <SettingsModal isOpen={isSettingsModalOpen} onClose={closeModal} initialTab={activeTab} />
```
[the remark this answers is not in the document]
Comment (Bhavyajain21) [reply src/components/Chat/StartChat.tsx]: Done!

File: src/components/Chat/StartChat.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (Bhavyajain21) [reply src/components/Chat/StartChat.tsx]: Done!

File: src/components/Chat/StartChat.tsx (hunk 4 above)
Comment (Bhavyajain21) [reply src/components/Chat/StartChat.tsx]: Done!

**Review by joseplayero — APPROVED** [review]
thanks for making changes. i added small change to make refresh always show :)

**Comment by Bhavyajain21** [comment]
Thanks for the tip! Much appreciated👏

**Comment by Bhavyajain21** [comment]
@joseplayero I'm open to contribute to more such issues. Please let me know if I can be of any help. Thanks!

**Pull request merged**